Language-Version-History.md

```diff
@@ -0,0 +1,68 @@
+Features Added in C# Language Versions
+====================
+
+# [C# 2](https://msdn.microsoft.com/en-us/library/7cz8t42e(v=vs.80).aspx) (VS 2005)
+- Generics
+- Partial types
+- Anonymous methods
+- Iterators
+- Nullable types
+- Getter/setter separate accessibility
+- Method group conversions (delegates)
+- Co- and Contra-variance for delegates and interfaces
+- Static classes
+- Delegate inference
+
+# [C# 3](https://msdn.microsoft.com/en-us/library/bb308966.aspx) (VS 2008)
+- Implicitly typed local variables
+- Object and collection initializers
+- Auto-Implemented properties
+- Anonymous types
+- Extension methods
+- Query expressions
+- Lambda expression
+- Expression trees
+- Partial methods
+
+# [C# 4](https://msdn.microsoft.com/en-us/magazine/ff796223.aspx) (VS 2010)
+- Dynamic binding
+- Named and optional arguments
+- Generic co- and contravariance
+- Embedded interop types ("NoPIA")
+
+# [C# 5](https://blogs.msdn.microsoft.com/mvpawardprogram/2012/03/26/an-introduction-to-new-features-in-c-5-0/) (VS 2012)
+- Asynchronous methods
+- Caller info attributes
+
+# [C# 6](https://github.com/dotnet/roslyn/wiki/New-Language-Features-in-C%23-6) (VS 2015)
+- [Specification online](https://github.com/dotnet/csharplang/blob/master/spec/README.md)
+- Compiler-as-a-service (Roslyn)
+- Import of static type members into namespace
+- Exception filters
+- Await in catch/finally blocks
+- Auto property initializers
+- Default values for getter-only properties
+- Expression-bodied members
+- Null propagator (null-conditional operator, succinct null checking)
+- String interpolation
+- nameof operator
+- Dictionary initializer
+
+# [C# 7.0](https://blogs.msdn.microsoft.com/dotnet/2017/03/09/new-features-in-c-7-0/) (VS 2017)
+- [Out variables](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.0/out-var.md)
+- [Pattern matching](https://github.com/dotnet/csharplang/blob/master/proposals/patterns.md)
+- [Tuples](https://github.com/dotnet/roslyn/blob/master/docs/features/tuples.md)
+- [Deconstruction](https://github.com/dotnet/roslyn/blob/master/docs/features/deconstruction.md) and [Discards](https://github.com/dotnet/roslyn/blob/master/docs/features/discards.md)
```

Comment:
Why grouped into the same bullet? It doesn't feel like deconstruction and discards have much in common, aside from alliteration.